src/librustc/diagnostics.rs

```diff
@@ -168,6 +168,29 @@ variant constructors or struct constructors (for unit or tuple structs). This
 is because Rust currently does not support compile-time function execution.
 "##,
 
+E0018: r##"
+The value of static and const variables must be known at compile time. You
+can't cast a pointer as an integer because we can't know what value the
+address will take.
+
+However, pointers to other constants' address are allowed in constants,
```

Comment:
hmm, grammatically, it needs to use "addresses" (plural) here, since you have an apostrophe after the "s".

That, or make it consistently singular, like so:

"However, a pointer to another constant's address is allowed in a constant. Example: ..."